docs/source/dev_guide/review_process.rst

```diff
@@ -0,0 +1,110 @@
+with any other added tests coming in from the updated master branch if they were
+not resolved in the merging or rebase process. If your PR started out as a
+Draft, this is now the time to create a normal PR. If the PR never went into a
+Draft PR this is the point where you would create create the PR. Upon creating
+the PR the continuous integration tools run a set of rigorous tests with
+different compilers and compiling modes such as debug to ensure compatibility
+with the our defined set of options.
+
+Once the PR is ready for review (all items on the checklist are completed), use
+the Github button to notify specific developers to review the changes.
+
+To help speed up the process we follow some rough guidelines so everyone knows
+what to expect from a reviewer's role:
+
+ * Once the reviewers' concerns have been addressed, this person is responsible
+   for approving them as soon as possible. This not only ensures that the
+   changes are available to everyone as soon as they are ready, but also
+   minimizes the need for further merges with master and associated extra work.
+
+ * Single comments without actually providing a review are allowed and
+   encouraged, but should be used sparsely. They can be really helpful if
+   something might be overseen. But if you want changes and involve yourself in
+   the process, make a proper review and take the responsibility to see it
+   through.
+
+ * If you are a reviewer and you start adding changes yourself (excluding the
+   one-line propositions that Github provides or just cosmetics) another person
+   should become the reviewer. This is to ensure that any part is reviewed by
+   someone else other than the creator.
+
+A last note related to our own experience. If it happens that your PR falls into
+a busy time with a lot of changes on the master branch which are unrelated to
+you, do not hesitate to contact us if the any issues arise *after* your PR has
+been approved. While we discourage changes after the PR has been approved,
+sometimes it is necessary to be able to merge with master. There are no rules to
+make this easy so please decide on the basis of the fact that we want a clean
+compiling master branch. To elaborate on this, here are a few examples
+
+ * A PR is approved but there are still changes added afterwards. While this can
+   not be avoided completely, it should not be the norm. As the PR requester and
+   if the changes are not only of cosmetic nature, as a courtesy please ask the
+   approving reviewer again for feedback on the changes.
+
+ * Changes due to merge or rebase with master because the master branch changed
+   can not be avoided. If the PR was approved, master changed and the result are
+   conflicts with the PR, this of course has to be solved. Again, as a courtesy,
+   please ask for feedback on the changes again because merges/rebases can be
+   complicated and we want to avoid a broken master branch.
+
+**TL;DR**
```

Comment:
Let's merge this with the `CONTRIBUTING.md`, which can serve as a short reference (it's shown to potential contributors on opening their first pull request). That document should, in turn, link to here for the full description.

Reply:
Yes, good idea. How do I make sure it is compiled into an html?
The links are hardcoded, right?

Reply:
The `CONTRIBUTING` file can link to the online HTML documentation (since it will need to be followable from github, effectively an external site).

Not sure about the link in the other direction, since the CONTRIBUTING file isn't compiled as part of the doc build. You could have a look at `index.rst` which includes sections of the `README` for one way to do it (if you just want to automatically paste the summary points to the bottom of the `review_process` page). It would also be good to just reference the file by name near the beginning of the document.